app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -177,7 +178,10 @@ const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(supercla
 
             // Process the metadata of each post
             if (post?.metadata && Object.keys(post?.metadata).length > 0) {
-                const data = post.metadata;
+                let data = post.metadata;
+
+                // parsing into json since notifications are sending metadata as a string
+                data = safeParseJSON(post.metadata) as PostMetadata;
```

Comment:
```suggestion
                // parsing into json since notifications are sending metadata as a string
                const data = safeParseJSON(post.metadata) as PostMetadata;
```